hal/shared/platforms.h

```diff
@@ -39,5 +39,40 @@
 #define PLATFORM_B5SOM                      25
 #define PLATFORM_NEWHAL                     60000
 
+#if PLATFORM_ID == PLATFORM_GCC
```

Comment:
Would per-platform `hal_platform_config.h` potentially be a better choice for this?